### ISSUE TITLE
Empty warning popup window when loading an invalid templates archive file

### ISSUE THREAD

**Issue by Infini-Creation** [godotengine/godot#85862]
### Tested versions

v4.2.stable.official [46dc27791]

### System information

Windows 10 (22H2 19045.3693)

### Issue description

When loading an invalid archive as export templates archive file, Godot editor open an empty warning pop-up window.
Note: "invalid" word means here a set of templates Godot didn't succeed to import, but it's a valid zip archive.

![Uploading godot-421-cb-exp-tpl-issue-emptywindow.png…]()

### Steps to reproduce

- create a zip archive with any stuff you have around like pictures
- rename file to any name with ".tpz" extension
- try to load it as template archive ("manage export templates" from menu then "install from file")
- the empty warning pop-up window is opened, no error or warning shown anywhere

### Minimal reproduction project (MRP)

N/A

**Comment by jsjtxietian** [comment]
I did some test, and find that actually any `EditorNode::get_singleton()->show_warning` behaves like this inside `ExportTemplateManager::_install_file_selected`.

**Comment by timothyqiu** [comment]
Changing `_install_file_selected` to `CONNECT_DEFERRED` seems to solve the issue.

```patch
diff --git i/editor/export/export_template_manager.cpp w/editor/export/export_template_manager.cpp
index 26ed7c46fb..26720b2d4a 100644
--- i/editor/export/export_template_manager.cpp
+++ w/editor/export/export_template_manager.cpp
@@ -993,7 +993,7 @@ ExportTemplateManager::ExportTemplateManager() {
 	install_file_dialog->set_file_mode(FileDialog::FILE_MODE_OPEN_FILE);
 	install_file_dialog->set_current_dir(EditorSettings::get_singleton()->get_meta("export_template_download_directory", ""));
 	install_file_dialog->add_filter("*.tpz", TTR("Godot Export Templates"));
-	install_file_dialog->connect("file_selected", callable_mp(this, &ExportTemplateManager::_install_file_selected).bind(false));
+	install_file_dialog->connect("file_selected", callable_mp(this, &ExportTemplateManager::_install_file_selected).bind(false), CONNECT_DEFERRED);
 	add_child(install_file_dialog);
 
 	hide_dialog_accept = memnew(AcceptDialog);
```

**Comment by jsjtxietian** [comment]
Yes it solves the issue, but I'm a little curious about why it breaks at the first place.

**Comment by neoascetic** [comment]
The problem in my case was missing "version.txt" in the archive. After I've added it I was at least able to install my custom export template.

**Comment by parsaiej** [comment]
Hi all, the linked PR should be merged. If a user follows the macOS documentation for building export templates, the user will hit a dead end due to this issue. Regarding the `version.txt`, shouldn't that be handled automatically by the `generate_bundle=yes` option?
